lab3/lab3_src.c

```diff
@@ -47,12 +47,13 @@ int main(void)
         sscanf(line, "%f", &load_avg);
 
         fclose(load_file);
+        
  
         /*
         * loop through pins and compare actual load average value to threshold
         * needed in order to turn pin on
         */
-        for (int pin = 0; pin < 8; pin++)
+        for (int pin = 0; pin < 7; pin++)
```

Comment:
Same issue here as with the array declaration, we would want the for loop to go 0-7 so in the second clause the condition should be pin < 8, so that the loop will stop once pin is larger than 7